src/systems/systems/src/air_conditioning/acs_controller.rs

```diff
@@ -0,0 +1,1808 @@
+        }
+        if matches!(
+            aircraft_state,
+            AcStateManager::BeginTakeOff(_)
+                | AcStateManager::EndTakeOff(_)
+                | AcStateManager::BeginLanding(_)
+                | AcStateManager::EndLanding(_)
+        ) {
+            intermediate_flow =
+                intermediate_flow.min(Ratio::new::<percent>(Self::FLOW_REDUCTION_LIMIT));
+        }
+        intermediate_flow.max(Ratio::new::<percent>(Self::BACKFLOW_LIMIT))
+    }
+
+    // This calculates the flow based on the demand, when the packs are modelled this needs to be changed
+    // so the demand actuates the valve, and then the flow is calculated based on that
+    fn absolute_flow_calculation(&self, pressurization: &impl CabinAltitude) -> MassRate {
+        let absolute_flow = self.flow_demand.get::<ratio>()
+            * (Self::FLOW_CONSTANT_XCAB * pressurization.cabin_altitude().get::<foot>()
+                + Self::FLOW_CONSTANT_C);
+        MassRate::new::<kilogram_per_second>(absolute_flow)
+    }
+
+    fn fcv_open_allowed_determination(&mut self, acs_overhead: &AirConditioningSystemOverhead) {
+        // Flow Control Valve 1
+        self.fcv_1_open_allowed = acs_overhead.pack_1_pb_is_on()
+            && !self.engine_1_in_start_mode
+            && (!self.engine_2_in_start_mode || !self.crossbleed_is_on)
+            && !self.engine_1_on_fire
+            && !self.ditching_is_on;
+        // && ! pack 1 overheat
+        // Flow Control Valve 2
+        self.fcv_2_open_allowed = acs_overhead.pack_2_pb_is_on()
+            && !self.engine_2_in_start_mode
+            && (!self.engine_1_in_start_mode || !self.crossbleed_is_on)
+            && !self.engine_2_on_fire
+            && !self.ditching_is_on;
+        // && ! pack 2 overheat
+    }
+
+    fn should_open_fcv_determination(&self, engines: [&impl EngineCorrectedN1; 2]) -> [bool; 2] {
+        // Engine bleed pushbuttons not yet integrated. TODO after Bleed PR is merged
+        [
+            self.fcv_1_open_allowed
+                && ((engines[0].corrected_n1() >= Ratio::new::<percent>(15.)
+                    || (engines[1].corrected_n1() >= Ratio::new::<percent>(15.)
+                        && self.crossbleed_is_on))
+                    || self.apu_bleed_on),
+            self.fcv_2_open_allowed
+                && ((engines[1].corrected_n1() >= Ratio::new::<percent>(15.)
+                    || (engines[0].corrected_n1() >= Ratio::new::<percent>(15.)
+                        && self.crossbleed_is_on))
+                    || self.apu_bleed_on),
+        ]
+    }
+}
+
+impl PackFlow for PackFlowController {
+    fn pack_flow(&self) -> MassRate {
+        if self.flow_control_valve_open {
+            if self.single_pack_operation {
+                self.absolute_flow
+            } else {
+                self.absolute_flow * 2.
+            }
+        } else {
+            MassRate::new::<kilogram_per_second>(0.)
+        }
+    }
```

Comment:
**suggestion**: Consider computing `pack_flow` once in the `update` function and storing it in a field. That way you can remove the `self.flow_control_valve_open`, `self.single_pack_operation`, and `self.absolute_flow` fields.

Reply:
Done! This simplified the code but maybe it made it a bit less readible.